enterprise/users/src/main/java/io/crate/auth/user/StatementPrivilegeValidator.java

```diff
@@ -559,5 +570,25 @@ public Void visitTableFunctionRelation(TableFunctionRelation tableFunctionRelati
         public Void visitQueriedSelectRelation(QueriedSelectRelation relation, RelationContext context) {
             return process(relation.subRelation(), context);
         }
+
+        @Override
+        public Void visitView(AnalyzedView analyzedView, RelationContext context) {
+            Privileges.ensureUserHasPrivilege(
+                context.type,
+                Privilege.Clazz.SCHEMA, // change to VIEW once we've View level permissions
+                analyzedView.name().schema(),
+                context.user
+            );
+            User owner = analyzedView.owner() == null ? null : userLookup.findUser(analyzedView.owner());
+            if (owner == null) {
+                // TODO: throw more detailed error referring to owner missing privileges?
```

Comment:
I don't think that throwing a detailed error here about owner's missing privileges would leak information here. so I'd be in favor of doing it.